ampcombi/clustering_hits.py

```diff
@@ -0,0 +1,158 @@
+#!/bin/python3
+
+# TITLE: Cluster the AMP hits using MMseqs2 after generating the complete summary
+
+import tempfile
+import shutil
+import subprocess
+import os
+import pandas as pd
+
+#########################################
+# FUNCTION: PARSE INPUT FILE
+#########################################
+def parsing_input_for_cluster(merged_df):
+    """
+    This parses the ampcombi file and prepares it for downstream processing by converting sequences to fasta files. 
+    """
+    # make a column for aa_sequnces headers
+    merged_df['seq_headers'] = merged_df['sample_id'] + '!' + merged_df['contig_id'] + '!' + merged_df['CDS_id']
+    # convert table to sequnecs in temporary folder
+    temp_folder = 'clusters'
+    os.makedirs(temp_folder, exist_ok=True)
+    #os.makedirs(temp_folder, exist_ok=True)
+    
+    fasta_entries = []
+    for index, row in merged_df.iterrows():
+        sequence_id = row['seq_headers']
+        sequence = row['aa_sequence']
+        fasta_entry = f'>{sequence_id}\n{sequence}\n'
+        fasta_entries.append(fasta_entry)
+    fasta_content = ''.join(fasta_entries)
+    # write to file 
+    output_file_path = os.path.join(temp_folder, 'merged_fastas.fasta')
+    with open(output_file_path, 'w') as fasta_file:
+        fasta_file.write(fasta_content)
+    return merged_df
+
+########################################
+# FUNCTION: CLUSTER USING MMSEQS2 
+#########################################
+def mmseqs_cluster(cov_mod,cluster_mode,coverage,seq_id,sensitivity,threads):
+    """
+    This clusters the sequences using mmseqs2 (please refer to https://mmseqs.com/latest/userguide.pdf).
+    - This mode of clustering : - combines prefiltering - alignment -clustering : so will take long to process. 
+                                - can choose single step or cascascade clustering please refer to (https://github.com/soedinglab/MMseqs2/wiki#clustering-databases-using-mmseqs-cluster-or-mmseqs-linclust) for more info
+                                - the clustering module can need large amounts of memory
+    - The coverage is bidirectiional query and target, so should have a coverage of 100% : --cov-mode 0 -c 1
+    - The sequence identity --min-seq-id 0.4
+    - The sensitivity can be changed with -s 7.5 (very sensitive) and 4 is fast (default)
```

Comment:
Sorry, I did not understand this sentence. Does it mean -s is set to 4 by default and this is fast, while it can be changed to 7.5 to very sensitive (presumably slow?)?

Reply:
exactly!